azkaban-common/src/main/java/azkaban/executor/selector/CandidateSelector.java

```diff
@@ -74,6 +74,10 @@ public K getBest(List<K> candidateList, V dispatchingObject) {
        return null;
      }
 
+     if (null == comparator){
+       logger.info("candidate comparator is not specified, default comparator from 'Collections' class will be used.");
```

Comment:
actually there is a  bug here, when the comparator passed over to the Collections.max function is null and the type of the object to select from doesn't implement Comparable the function will throw exception instead of using any default comparators to get the job done, very nice finding,  fixed by restricting the object type to extend from Comparable.